File: app/src/main/java/com/returnnotfound/stain/di/module/ApiServiceModule.java
package com.returnnotfound.stain.di.module;

import com.returnnotfound.stain.data.remote.AnimeService;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import retrofit2.Retrofit;

@Module
public abstract class ApiServiceModule {
  @Provides
  @Singleton
  static AnimeService provideAnimeService(Retrofit retrofit) {
    return retrofit.create(AnimeService.class);
  }
}

File: app/src/main/java/com/returnnotfound/stain/di/module/ApiServiceModule.kt
package com.returnnotfound.stain.di.module

import com.returnnotfound.stain.data.remote.AnimeService
import dagger.Module
import dagger.Provides
import retrofit2.Retrofit
import javax.inject.Singleton

@Module
object ApiServiceModule {
  @JvmStatic
  @Provides
  @Singleton
  fun provideAnimeService(retrofit: Retrofit): AnimeService {
    return retrofit.create(AnimeService::class.java)
  }
}